### PR TITLE
C++ STL Algorithms Series - Part 2 Blog

### PR DESCRIPTION
**Pull request by aFinchy**
## Change summary

Added blog C++ STL Algorithms Series - Part 2

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [ ] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/blog/posts/c-stl-algorithms-series-part-2.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Topic
+
+Continuing from the previous entry in this series, we’re going to keep exploring some of the use-cases of `<algorithm>/<numeric>` in our day to day work.
+
+## Example
+
+It’s all too easy to come up with trivial examples for algorithms that don’t reflect the type of problems we have to solve everyday. With that in mind, this example (though a little contrived) might be something we actually want to calculate in O3DE.
+
+Suppose we have a collection of Aabbs and we’d like to find the union of all of them. We might need this to find the total extents of our scene or perhaps the area taken up by a players base. Here’s the normal imperative version to start:
```
Comment (micronAMZN) [inline content/blog/posts/c-stl-algorithms-series-part-2.md:19]: suggest using the acronym style guidance here.

```suggestion
Suppose we have a collection of axis aligned bounding boxes (AABBs) and we’d like to find the union of all of them. We might need this to find the total extents of our scene or perhaps the area taken up by a players base. Here’s the normal imperative version to start:
```

File: content/blog/posts/c-stl-algorithms-series-part-2.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+The talk does a fantastic job of introducing a whole host of algorithms and how to think about them.
+
+## To be continued...
+
+Next time we’ll look at another algorithm staple - `transform`.
+
+Author: Tom Hulton-Harrop
+
+_Disclaimer: The views expressed here are those of the individual author and do not represent those of Game Engine and Developer Services or Amazon._
```
Comment (micronAMZN) [inline content/blog/posts/c-stl-algorithms-series-part-2.md:69]: Same comment as the PR for Part 1 regarding disclaimers.

**Review by micronAMZN — APPROVED** [review]
Approving with a minor suggestion and note about the disclaimer.

File: content/blog/posts/c-stl-algorithms-series-part-2.md (hunk 2 above)
Comment (aFinchy) [reply content/blog/posts/c-stl-algorithms-series-part-2.md:69]: I will add the new one. The new one will be 
Disclaimer: The views expressed here are those of the individual author and do not represent those of the Open 3D Foundation, Open 3D Engine or individual's respective company.

**Review by hultonha — APPROVED** [review]
I'm super biased but approving anyway 😅 Thanks so much for getting these published @aFinchy

File: content/blog/posts/c-stl-algorithms-series-part-2.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Deliberation
+
+It’s debatable if the algorithm version is more succinct than the first, but it does have some advantages. First, the return value can now be `const`, as everything is calculated in a single statement. Eliminating mutable state is well understood to reduce the likelihood of bugs creeping into code. The lambda is a pure function and due to having no capture (highly recommended) we can quickly determine it has no side effects. The other advantage is the name, we can glance at the code and quickly get an intuition for what operation is taking place (even if we have to check the types and lambda to understand all the details).
+
+Applications of `accumulate` and `transform` (look out for this in an upcoming article) crop up all over the place and making use of them can help simplify and improve our code. (Note: `accumulate` is very closely related to `reduce` - they are largely the same but do have some differences when it comes to ordering and execution policies. This isn’t important for understanding what they actually do).
+
+## Further Reading
+
+If you’re getting more interested in algorithms the next must-watch talk is
```
Comment (FiniteStateGit) [inline content/blog/posts/c-stl-algorithms-series-part-2.md:57]:
```suggestion
If you’re getting more interested in algorithms the next must-watch talk is:
```

**Pull request merged**